pkgs/tools/security/wordlists/default.nix

```diff
@@ -0,0 +1,78 @@
+{ lib
+, runtimeShell
+, stdenv
+, tree
+, pkgs
+}:
+
+let
+  _wordlistData =
+    [ { name = "dirbuster"; expressionPath = ./dirbuster.nix; }
+      { name = "nmap"; expressionPath = ./nmap.nix; needsPackage = true; }
+      { name = "rockyou"; expressionPath = ./rockyou.nix; }
+      { name = "seclists"; expressionPath = ./seclists.nix; }
+      { name = "wfuzz"; expressionPath = ./wfuzz.nix; needsPackage = true; }
+    ];
+
+  scopedWordlists = lib.makeScope pkgs.newScope (self: with self;
+    let _packageWithLinkCommandFrom = acc: datum:
+      # Some wordlists are based on an already existing `nixpkgs` package; Pass it as parameter of the expression.
+      let parameters = {}
+        // lib.optionalAttrs ((builtins.elem "needsPackage" (builtins.attrNames datum)) && datum.needsPackage) { ${datum.name} = pkgs.${datum.name}; };
```

Comment:
```suggestion
    let _packageWithLinkCommandFrom = acc: datum@{ needsPackage ? false, ... }:
      # Some wordlists are based on an already existing `nixpkgs` package; Pass it as parameter of the expression.
      # I'm not sure if the `inherit` usage below works as expected, but should be tried.
      let parameters = lib.optionalAttrs needsPackage { inherit (pkgs) "${datum.name}"; };
```

Reply:
Sadly this is illegal!

```nix
error: --- ParseError ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ nix-shell
at: (21:72) in file: /home/pamplemousse/Workspace/tools/nixpkgs/pkgs/tools/security/wordlists/default.nix

    20|       # I'm not sure if the `inherit` usage below works as expected, but should be tried.
    21|       let parameters = lib.optionalAttrs needsPackage { inherit (pkgs) "${datum.name}"; };
      |                                                                        ^

dynamic attributes not allowed in inherit
```

Reply:
ok, the rest of the suggestion still applies:
* `{} //` is unnecessary
* the whole builtin mashup can be replaced by extracting out of the argument

Reply:
```suggestion
    let _packageWithLinkCommandFrom = acc: datum@{ needsPackage ? false, ... }:
      # Some wordlists are based on an already existing `nixpkgs` package; Pass it as parameter of the expression.
      let parameters = lib.optionalAttrs needsPackage { ${datum.name} = pkgs.${datum.name}; };
```

Reply:
remains to be fixed.

Reply:
still not fixed.